<?xml version="1.0"?>
<launch>

  <arg name="cf_num" default="1" />
  <arg name="z_coefficient" default="4"/>
  <arg name="continuous" default="True" />
  <arg name="air_vel" default=".3" />
  <arg name="land_vel" default=".15" />
  <arg name="takeoff_vel" default=".15" />
  <arg name="landing_vel" default=".3" />
  <arg name="follow" default="True" />
  <arg name="follow_nodes" default="cbs" />


  <remap from="easy_sg/StartingID_topic" to="setup/StartingID_topic" />
  <remap from="si_planner/Starter" to="setup/Starter" />

  <node name="si_planner" pkg="planner" type="si_planner.py" output="screen">
  	<param name="cf_num" value="$(arg cf_num)"/>
  	<param name="z_coefficient" value="$(arg z_coefficient)" />
  	<param name="continuous" value="$(arg continuous)" />
  	<param name="land_vel" value="$(arg land_vel)" />
  	<param name="air_vel" value="$(arg air_vel)" />
    <param name="takeoff_vel" value="$(arg takeoff_vel)" />
    <param name="landing_vel" value="$(arg landing_vel)" />
    <param name="air_buffer_dist" value=".5" />
    <param name="buffer_z_frac" value=".5" />
    <param name="show_reserved" value="False" />
  </node>

  <node name="easy_sg" pkg="planner" type="easy_sg.py">
    <param name="follow" value="$(arg follow)" />
    <rosparam command="load" file="$(find planner)/config/$(arg follow_nodes).yaml" />
  </node>

  <node name="setup" pkg="planner" type="setup.py" output="screen">
    <param name="cf_num" value="$(arg cf_num)" />
    <param name="centered" value="True" />
    <param name="follow" value="$(arg follow)" />
    <rosparam command="load" file="$(find planner)/config/$(arg follow_nodes).yaml" />
  </node>

</launch>

